jupiterone/resource_user_group_membership_test.go

```diff
@@ -0,0 +1,183 @@
+package jupiterone
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/jupiterone/terraform-provider-jupiterone/jupiterone/internal/client"
+)
+
+func TestUserGroupMembership_Basic(t *testing.T) {
+	ctx := context.TODO()
+
+	recordingClient, directClient, cleanup := setupTestClients(ctx, t)
+	defer cleanup(t)
+
+	resourceName := "jupiterone_user_group_membership.test_user"
+	userGroupName := "tf-provider-test-user-group"
+	email := "test.user@jupiterone.com"
+
+	resource.Test(t, resource.TestCase{
```

Comment:
Do we need to use this terraform test resource?  What does it afford us over the built in tester?

Reply:
I have no better explanation other than that is how it was done here before me. I think I'll keep it as is and explore this question more when I get more comfortable in this repo. Thank you for calling it out.